react-game/src/App.jsx

```diff
@@ -0,0 +1,195 @@
+import React, { useState, useEffect } from 'react';
+import useSound from 'use-sound';
+import QuestionCard from './Components/Questions/QuestionCard';
+import fetchQuizQuestions from './API';
+import './App.css';
+import btnSound from './music/btn.mp3';
+import { menu } from './styles';
+import Statistics from './Components/Statistics/Statistics';
+import SoundSettings from './Components/SoundSettings/SoundSettings';
+import GameSettings from './Components/GameSettings/GameSettings';
+
+const App = () => {
+  const [loading, setLoading] = useState(false);
+  const [questions, setQuestions] = useState([]);
+  const [number, setNumber] = useState(0);
+  const [userAnswers, setUserAnswers] = useState([]);
+  const [score, setScore] = useState(0);
+  const [gameOver, setGameOver] = useState(true);
+  const [difficulty, setDifficulty] = useState('easy');
+  const [category, setCategory] = useState('9');
+  const [settings, setSettings] = useState(false);
+  const [amount, setAmount] = useState(10);
+  const [statistic, setStatistic] = useState(false);
+  const [soundSettings, setSoundSettings] = useState(false);
+
+  const [volume, setVolume] = useState(0.75);
+
+  const setKey = (event) => {
+    if (event.key === ' ') {
+      const btn = document.querySelector('#next');
+      if (btn !== null) {
+        btn.click();
+      }
+    }
+  };
+
+  useEffect(() => {
+    document.addEventListener('keydown', setKey);
+    return () => {
+      window.removeEventListener('keydown', setKey);
+    };
+  }, []);
+
+  const [play] = useSound(
+    btnSound,
+    {
+      volume,
+      interrupt: true,
+    },
+  );
+
+  const startTrivia = async () => {
+    setLoading(true);
+    setGameOver(false);
+    console.log(amount,
+      difficulty,
+      category,
+      amount);
+
+    const newQuestions = await fetchQuizQuestions(
+      amount,
+      difficulty,
+      category,
+      amount,
+    );
+    setQuestions(newQuestions);
+    setScore(0);
+    setUserAnswers([]);
+    setNumber(0);
+    setLoading(false);
+  };
+
+  const checkAnswer = (e) => {
+    play();
+    if (!gameOver) {
+      const answer = e.currentTarget.value;
+
+      const correct = questions[number].correct_answer === answer;
+
+      const item = JSON.parse(localStorage.getItem(category));
+
+      if (correct) {
+        setScore((prev) => prev + 1);
```

Comment:
no need to you callback inside set state here. 

```suggestion
        setScore(score + 1);
```
should be enough. but god that you know about it.